packages/domain/src/use-cases/publications/Reaction.ts

```diff
@@ -23,24 +23,12 @@ export class Reaction {
   ) {}
 
   async add(request: ReactionRequest) {
+    await this.gateway.add(request);
     await this.presenter.add(request);
-
-    try {
-      await this.gateway.add(request);
-    } catch (e) {
-      await this.presenter.remove(request);
```

Comment:
the problem with this approach was, that when another type of reaction already exists, and we try to add a new one, then gateway call fails, and we don't know the previous state to revert to (could be `reaction:null` or `reaction:another type than what we tried to add`) 

so, as reactions are stored in API only, the request is fast, I decided we don't need optimistic update here, otherwise we need to store somewhere the previous value.

Reply:
This could be tricky on slower networks, imagine someone on 3g who could visibly see the lag when pressed.
I live it for @cesarenaldi to decide if we want that to be fully optimistical or allow some sort of delay and force frontends to deal with `loading` state

Reply:
yes, good point, I think it's better to keep it as it was then, and just disable button in the UI, reactions need to be redone from the API side anyways